### PR TITLE
Shrink hello world by 3.2%

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
* Get rid of the MethodTable for double/float
* Get rid of any `Array<T>` methods

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs [hunk 1]
```diff
@@ -43,15 +43,15 @@ static class NativeFormatEnumInfo
                     var handle = field.DefaultValue;
                     unsortedBoxedValues[i] = handle.HandleType switch
                     {
-                        HandleType.ConstantSByteValue => (byte)handle.ToConstantSByteValueHandle(reader).GetConstantSByteValue(reader).Value,
+                        HandleType.ConstantSByteValue => (object)(byte)handle.ToConstantSByteValueHandle(reader).GetConstantSByteValue(reader).Value,
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs:46]: This was a fun headscratcher.

**Review by jkotas — APPROVED** [review]
Nice!

File: src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs (hunk 1 above)
Comment (VSadov) [reply src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs:46]: What was going on here?

Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs:46]: type inference not cooperating

Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfo.cs:46]: Yup, things got widened to uint64 and then boxed as uint64. We need to box before widening.

**Pull request merged**